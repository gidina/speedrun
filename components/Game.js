import React from "react";
import PropTypes from 'prop-types';
import Link from 'next/link';

import "./index.css";

const Game = props => {
    const { game } = props;
    return (
        <li className="game">
            <img alt="" className="game-img" src={game.logoUrl} />
            <Link as={`/game/${game.id}`} href={`/game?id=${game.id}`}>
                <a className="game-title">{game.name}</a>
            </Link>
        </li>
    );
};

export default Game;

Game.propTypes = {
    game: PropTypes.shape({
        id: PropTypes.integer,
        logoUrl: PropTypes.string,
        name: PropTypes.string
    }).isRequired
};