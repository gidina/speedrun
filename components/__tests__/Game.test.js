import React from 'react';
import renderer from 'react-test-renderer';
import Game from '../Game';

describe("Game component", () => {
    it('Renders correctly', () => {
        const game = {
            id: "game-1", 
            name: "international-name",
            logoUrl: "cover-small-uri"
        };
        const tree = renderer
            .create(<Game game={game} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});