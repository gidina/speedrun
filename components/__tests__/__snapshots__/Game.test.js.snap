// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component Renders correctly 1`] = `
<Link-mock
  as="/game/game-1"
  href="/game?id=game-1"
>
  <li
    className="game"
  >
    <img
      alt=""
      className="game-img"
      src="cover-small-uri"
    />
    <h3
      className="game-title"
    >
      international-name
    </h3>
  </li>
</Link-mock>
`;
